ppl/zqd/temporal/spaceworkflow.go

```diff
@@ -0,0 +1,100 @@
+package temporal
+
+import (
+	"context"
+	"time"
+
+	"github.com/brimsec/zq/api"
+	"github.com/brimsec/zq/ppl/zqd/apiserver"
+	"go.temporal.io/sdk/activity"
+	"go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+)
+
+var config Config
+var manager *apiserver.Manager
+
+func InitSpaceWorkflow(c Config, m *apiserver.Manager, r worker.Registry) {
+	config = c
+	manager = m
+	r.RegisterWorkflow(spaceWorkflow)
+	r.RegisterActivity(spaceCompactActivity)
+	r.RegisterActivity(spacePurgeActivity)
+}
+
+const writeSignal = "write"
+
+func spaceWorkflow(ctx workflow.Context) error {
+	logger := workflow.GetLogger(ctx)
+	logger.Info("Workflow started")
+	signalCh := workflow.GetSignalChannel(ctx, writeSignal)
+	for {
+		logger.Info("Workflow waiting for signal")
+		var signalVal string
+		signalCh.Receive(ctx, &signalVal)
+
+		logger.Info("Workflow waiting for timer or signal")
+		s := workflow.NewSelector(ctx)
+		timerCtx, cancel := workflow.WithCancel(ctx)
+		timerFuture := workflow.NewTimer(timerCtx, config.SpaceCompactDelay)
+		var err error
+		var executedActivity bool
+		s.AddFuture(timerFuture, func(f workflow.Future) {
```

Comment:
```suggestion
		s.AddFuture(timerFuture, func(_ workflow.Future) {
```

Reply:
Done in 3591e5cf43e6699497dfb8b7fb8bee1a207e8115.